ra/registration-authority.go

```diff
@@ -602,6 +608,12 @@ func (ra *RegistrationAuthorityImpl) UpdateRegistration(base core.Registration,
 
 // UpdateAuthorization updates an authorization with new values.
 func (ra *RegistrationAuthorityImpl) UpdateAuthorization(base core.Authorization, challengeIndex int, response core.Challenge) (authz core.Authorization, err error) {
+	// Refuse to update expired authorizations
+	if base.Expires == nil || base.Expires.Before(ra.clk.Now()) {
+		err = fmt.Errorf("Expired authorization")
```

Comment:
I hadn't thought of the race issue before just now. Based on that, I think it makes sense to consolidate the WFE check into RA and have the RA return core.NotFoundError.

Relatedly, you and I talked a little while ago about NotFound (404) vs Gone (410). You expressed a preference for Gone, and I had no strong preference. After thinking about it some more, my preference is now NotFound. My reasoning is: If we use Gone, the authorization resource goes through two transitions: OK -> Gone, and then later Gone -> NotFound when we delete the authorization object. Since the latter transition is administrative and not meaningful to the user, I'd rather just go straight from OK -> NotFound.